### PR TITLE
VZ-9204: Support for SMTP alert configuration in Grafana

### PR DESCRIPTION
**Pull request by lake-of-dreams**
see https://github.com/verrazzano/verrazzano-monitoring-operator/pull/348 for corresponding changes in verrazzano-monitoring-operator
CR and logic changes for configuring SMTP channels/notifications in Grafana in Verrazzano

### CODE REVIEW

**Review by michael-cico — DISMISSED** [review]

File: platform-operator/helm_config/charts/verrazzano-monitoring-operator/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml [hunk 1]
```diff
@@ -433,6 +433,59 @@ spec:
...
+                      passwordKey:
+                        description: The key in the existing SMTP secret containing
+                          the password.
+                        type: string
+                      skipVerify:
+                        description: If true, do not Verify SSL for SMTP server.
+                        type: boolean
+                      startTLSPolicy:
+                        description: Either “OpportunisticStartTLS”, “MandatoryStartTLS”,
+                          “NoStartTLS”. Default is empty.
```
Comment (kminder) [inline platform-operator/helm_config/charts/verrazzano-monitoring-operator/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:482]: What behavior does the default empty value result in?

Comment (lake-of-dreams) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/crds/verrazzano.io_verrazzanomonitoringinstances_crd.yaml:482]: empty value results in `MandatoryStartTLS`
Ref:
https://github.com/grafana/grafana/blob/afc9925dbfa55886cc85a9a2106ed9892d7df239/pkg/services/notifications/smtp.go#L144 returns 0 which translates to `MandatoryStartTLS` in https://github.com/go-mail/mail/blob/f59b9b83a4e522098e3d3eb94e6f81850ad6e973/smtp.go#L167

File: THIRD_PARTY_LICENSES.txt [hunk 2]
```diff
@@ -5088,5 +5088,5 @@ The above copyright notice and either this complete permission notice or at a mi
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 === ATTRIBUTION-HELPER-GENERATED:
-=== Attribution helper version: {Major:0 Minor:11 GitVersion:0.10.0-108-gc398c3cf GitCommit:c398c3cf1bca9bc72e2382e8918b1e9ccedd0816 GitTreeState:clean BuildDate:2023-04-16T22:18:59Z GoVersion:go1.19.2 Compiler:gc Platform:linux/amd64}
```
[the remark this answers is not in the document]
Comment (lake-of-dreams) [reply THIRD_PARTY_LICENSES.txt:5091]: fixed

File: platform-operator/controllers/verrazzano/component/grafana/vmi_test.go [hunk 3]
```diff
@@ -27,6 +28,20 @@ var grafanaEnabledCR = vzapi.Verrazzano{
 					Host: "testhost:3032",
 					Name: "grafanadb",
 				},
+				SMTP: &vmov1.SMTPInfo{
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzano/component/grafana/vmi_test.go:31]: The doc should make it clear as to which fields are mandatory and which are optional.  Maybe a link to SMTP doc is good enough.

**Review by pfmackin — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/grafana/vmi_test.go (hunk 3 above)
Comment (lake-of-dreams) [reply platform-operator/controllers/verrazzano/component/grafana/vmi_test.go:31]: ok will create doc pr separately

**Review by bhatpmk — APPROVED** [review]

**Pull request merged**